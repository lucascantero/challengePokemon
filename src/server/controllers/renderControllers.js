import React from 'react';
import ReactDom from 'react-dom/server';
import App from '../../client/app.jsx';
import {StaticRouter} from 'react-router-dom';

const renderControllers = function (req, res,next) {

    try {

        const componentString = ReactDom.renderToString(<StaticRouter location={req.url}> <App/> </StaticRouter>);

        const root = `
            <html lang="en">
                <head>
                    <meta charset="UTF-8">
                    <meta name="viewport" content="width=device-width, initial-scale=1.0">
                    <link rel="stylesheet" href="/public/dist/static/app.css">
                    <title>App Test</title>            
                </head>
                <body>   
                    <div id="root">${componentString}</div>
                    <script src='/public/dist/static/app.js'></script>
                </body>
            </html>
        `

        res.send(root)
    }
    catch (err){  
        next(err);
    }
}

export default renderControllers;
