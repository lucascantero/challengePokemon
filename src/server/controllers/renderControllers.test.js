import renderControllers from './renderControllers.js';

describe('Testing renderControllers', () => {

    it('Should match snapshot', () => {
        
        const req = {url: '/'};
        const res = {send: jest.fn()}
    

        renderControllers(req,res);

        const html = res.send.mock.calls[0][0];

        expect(html).toMatchSnapshot(); 

    });

});