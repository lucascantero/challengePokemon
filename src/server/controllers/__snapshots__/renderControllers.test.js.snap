// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing renderControllers Should match snapshot 1`] = `
"
            <html lang=\\"en\\">
                <head>
                    <meta charset=\\"UTF-8\\">
                    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
                    <link rel=\\"stylesheet\\" href=\\"/public/dist/static/app.css\\">
                    <title>App Test</title>            
                </head>
                <body>   
                    <div id=\\"root\\"> <div class=\\"searchBox__contenedor\\"><div class=\\"buttonLanguage__wrapper\\"><button class=\\"buttonLanguage__button\\">Inglés</button><button class=\\"buttonLanguage__button\\">Español</button></div></div><div class=\\"home__contenedor\\"><div class=\\"home__wrapper\\"><ul class=\\"home__ul\\"></ul><div class=\\"home__groupButton\\"><button class=\\"home__button\\" disabled=\\"\\">Anterior</button><button class=\\"home__button\\">Siguiente</button></div></div></div> </div>
                    <script src='/public/dist/static/app.js'></script>
                </body>
            </html>
        "
`;
